### DISCUSSION TITLE
seata AT模式支持回滚Redis操作吗？

### DISCUSSION THREAD

**Discussion by lvluogang** [apache/incubator-seata#6036] [Q&A]
业务流程中向MySQL数据库insert数据时，同时将数据缓存到Redis中，请问AT模式下如何同时回滚Redis缓存操作，将数据从Redis中删除？

**Comment by wangliang181230** [comment]
我们之前讨论过，大致有2种方案：
1. 把写缓存放到全局事务二阶段提交时，再写，二阶段提交时，去数据库读取数据，再写缓存。
     缺点：缓存会有延迟；
     需考虑的问题：对于其他全局事务二阶段未提交时，直接去数据库读数据，会读到脏数据。（如果你只是针对insert后写缓存的场景，那这个问题不存在。）
2. 一阶段，先变更原缓存key，再写新缓存；二阶段提交时，删除原缓存即可；二阶段回滚时，删除缓存，并回滚缓存。
    缺点：并发回滚时，很难判断哪个备份的原缓存，才是正确的。当然，也可以不去备份原缓存，直接从数据库里读取当前的数据，进行缓存也可以。
